core/src/main/scala/org/scalafmt/util/TokenOps.scala

```diff
@@ -65,12 +65,11 @@ object TokenOps {
   }
 
   def isTopLevelStatment(tok: Token, owner: Tree): Boolean = {
-      tok.is[KwObject] ||
-        tok.is[KwDef] ||
-        tok.is[KwCase] ||
-        tok.is[KwPackage] ||
-        owner.parent.exists(_.is[Defn.Class]) ||
-        owner.parent.exists(_.is[Defn.Trait])
+    tok.is[KwObject] || owner.parent.exists(_.is[Defn.Object]) ||
```

Comment:
Is the ` || ` part necessary? I think a match against `owner` might be sufficient.